src/analysis/normed_space/dual.lean

```diff
@@ -96,3 +117,160 @@ end
 end bidual_isometry
 
 end normed_space
+
+namespace inner_product_space
+open is_R_or_C continuous_linear_map
+
+variables (𝕜 : Type*)
+variables {E : Type*} [is_R_or_C 𝕜] [inner_product_space 𝕜 E]
+local notation `⟪`x`, `y`⟫` := @inner 𝕜 E _ x y
+local postfix `†`:90 := @is_R_or_C.conj 𝕜 _
+
+/--
+Given some `x` in an inner product space, we can define its dual as the continuous linear map
+`λ y, ⟪x, y⟫`.
+-/
+def to_dual' (x : E) : normed_space.dual 𝕜 E :=
+linear_map.mk_continuous
+  { to_fun := λ y, ⟪x, y⟫,
+    map_add' := λ _ _, inner_add_right,
+    map_smul' := λ _ _, inner_smul_right }
+  ∥x∥
+  (λ y, by { rw [is_R_or_C.norm_eq_abs], exact abs_inner_le_norm _ _ })
+
+@[simp] lemma to_dual'_apply {x y : E} : to_dual' 𝕜 x y = ⟪x, y⟫ := rfl
+
+/-- In an inner product space, the norm of the dual of a vector `x` is `∥x∥` -/
+@[simp] lemma to_dual'_isometry (x : E) : ∥to_dual' 𝕜 x∥ = ∥x∥ :=
+begin
+  refine le_antisymm _ _,
+  { exact linear_map.mk_continuous_norm_le _ (norm_nonneg _) _ },
+  { cases eq_or_lt_of_le (norm_nonneg x) with h h,
+    { have : x = 0 := norm_eq_zero.mp (eq.symm h),
+      simp [this] },
+    { refine (mul_le_mul_right h).mp _,
+      calc ∥x∥ * ∥x∥ = ∥x∥ ^ 2 : by ring
+      ... = re ⟪x, x⟫ : norm_sq_eq_inner _
+      ... ≤ abs ⟪x, x⟫ : re_le_abs _
+      ... = ∥to_dual' 𝕜 x x∥ : by simp [norm_eq_abs]
+      ... ≤ ∥to_dual' 𝕜 x∥ * ∥x∥ : le_op_norm (to_dual' 𝕜 x) x } }
+end
+
+variables {F : Type*} [inner_product_space ℝ F]
+
+/-- In an inner product space `F`, the function that takes a vector `x` in `F` to its dual
+`λ y, ⟪x, y⟫` is a continuous linear map. -/
+-- TODO extend to `is_R_or_C` (requires a definition of conjugate linear maps)
+def to_dual_map : F →L[ℝ] (normed_space.dual ℝ F) :=
+linear_map.mk_continuous
+  { to_fun := to_dual' ℝ,
+    map_add' := λ x y, by { ext, simp [inner_add_left] },
+    map_smul' := λ c x, by { ext, simp [inner_smul_left] } }
+  1
+  begin
+    intros x,
+    apply op_norm_le_bound,
+    { simp [norm_nonneg] },
+    { intros y,
+      simp only [one_mul, linear_map.coe_mk, to_dual'_apply, norm_eq_abs, abs_inner_le_norm] }
+  end
+
+@[simp] lemma to_dual_map_apply {x y : F} : to_dual_map x y = ⟪x, y⟫_ℝ := rfl
+
+/-- In an inner product space, the norm of the dual of a vector `x` is `∥x∥` -/
+@[simp] lemma to_dual_map_isometry (x : F) : ∥to_dual_map x∥ = ∥x∥ := to_dual'_isometry _ _
+
+lemma to_dual_map_injective : (@to_dual_map F _).ker = ⊥ :=
```

Comment:
This lemma should be named `ker_to_dual_map` and it should be a `simp` lemma. Please also add `to_dual_map_injective : injective (@to_dual_map F _) := to_dual_map_isometry.injective`. Then you can use https://leanprover-community.github.io/mathlib_docs/linear_algebra/basic.html#linear_map.ker_eq_bot to get `ker_to_dual_map`. I'd add a version of `ker_eq_bot_of_injective` or `ker_eq_bot` in the `continuous_linear_map` namespace too.

Reply:
I don't follow this comment -- the proof really proceeds by showing the kernel is zero, not by first proving it's injective and deducing as a corollary that the kernel is zero.  Right?

Reply:
We can get injectivity for free from `isometry to_dual_map`, then the proof of `ker to_dual_map = \bot` will occupy 1 line.